cmd/tetra/common/client.go

```diff
@@ -57,7 +56,7 @@ func connect(ctx context.Context) (*grpc.ClientConn, string, error) {
 	//       context anyway.
 	//       If that address is set try it, if it fails for any reason then retry
 	//       last time with the server address.
-	if viper.IsSet(KeyServerAddress) == false {
+	if ServerAddress == "" {
```

Comment:
aaaah maybe you removed the default value because of this. I think @tixxdz worked on this, what do you think? Should we remove the default value `localhost:54321` from the flag, see this https://github.com/cilium/tetragon/pull/1119/files#r1379864446.

Reply:
sorry to delay this PR again @Jack-R-lantern, will just need a look from Djalal because he worked on the default gRPC address to put so let him just ack this :)

Reply:
as far as I understand it would not be okay to remove the default since for most `tetra` users, tetragon might not run locally and the file (tetragon-info.json) it's trying to read will not be available. But with the default value, this part of the code will never be triggered. Maybe we could try to read the file containing the server address and fall back to the default `localhost:54321` value on file read error?

Reply:
this part of the code was never reached in practice because `IsSet` returns true even if it comes from the default value apparently. See https://github.com/spf13/viper/issues/323

Reply:
I haven't tested it but I think `viper.IsSet` returned false for the default value. So we would need to test this and to get the same behavior as before we should change some stuff here.

Reply:
I'll do some more thinking about that code as well

Reply:
@tixxdz could you take a look at this one :)?